packages/dev/core/src/Collisions/gpuPicker.ts

```diff
@@ -295,117 +323,279 @@ export class GPUPicker {
             return Promise.resolve(null);
         }
 
+        const { x: adjustedX, y: adjustedY, rttSizeW, rttSizeH } = this._prepareForPicking(x, y);
+        if (adjustedX < 0 || adjustedY < 0 || adjustedX >= rttSizeW || adjustedY >= rttSizeH) {
+            return Promise.resolve(null);
+        }
+
+        this._preparePickingBuffer(this._engine!, rttSizeW, rttSizeH, x, y);
+
+        const result = await this._executePicking(adjustedX, adjustedY, rttSizeW, rttSizeH, disposeWhenDone);
+        return result;
+    }
+
+    /**
+     * Execute a picking operation on multiple coordinates
+     * @param xy defines the X,Y coordinates where to run the pick
+     * @param disposeWhenDone defines a boolean indicating we do not want to keep resources alive (false by default)
+     * @returns A promise with the picking results. Always returns an array with the same length as the number of coordinates. The mesh or null at the index where no mesh was picked.
+     */
+    public async multiPickAsync(xy: { x: number; y: number }[], disposeWhenDone = false): Promise<Nullable<IGPUMultiPickingInfo>> {
+        if (!this._pickableMeshes || this._pickableMeshes.length === 0 || xy.length === 0) {
+            return Promise.resolve(null);
+        }
+
+        if (xy.length === 1) {
+            const pi = await this.pickAsync(xy[0].x, xy[0].y, disposeWhenDone);
+            Promise.resolve({
+                meshes: [pi?.mesh],
+                thinInstanceIndexes: [pi?.thinInstanceIndex],
+            });
+        }
+
+        const { minX, minY, maxX, maxY } = this._getBoundingBox(xy);

```

Comment:
I'm seeing quite a few GC that could be avoided. We are creating a lot of temp objects here (xy for instance or this const structures here

Reply:
Thanks, let me run one more lap with GC in mind.

Reply:
did anything change after this comment?

Reply:
https://github.com/BabylonJS/Babylon.js/pull/15423/commits/16bb98c610d5150cb51c9715a87b107bf967731a